extensions/amp-story-auto-ads/0.1/amp-story-auto-ads.js

```diff
@@ -449,8 +455,18 @@ export class AmpStoryAutoAds extends AMP.BaseElement {
       }
     }
 
-    userAssert(!!ALLOWED_AD_TYPES[configAttrs.type], `${TAG}: ` +
-      `"${configAttrs.type}" ad type is not supported`);
+    const {type} = configAttrs;
+    userAssert(!!ALLOWED_AD_TYPES[type], `${TAG}: ` +
+      `"${type}" ad type is not supported`);
+
+    if (type === 'fake') {
+      const id = this.element.getAttribute('id');
+      if (!id || !startsWith(id, 'i-amphtml-demo-')) {
+        user().warn(TAG, 'id must start with i-amphtml-demo- to use fake ads');
+        return null;
```

Comment:
Have we handle the case where `this.createAdElement_` returns null?
https://github.com/ampproject/amphtml/blob/391a431816894cc726e8c7891081885589f391bf/extensions/amp-story-auto-ads/0.1/amp-story-auto-ads.js#L356

Reply:
Nope you are right, null check was in wrong place. Instead of adding a bunch of logic for fake ad destruction, what do you think about creating a dummy element instead like `<p>Fake ad discarded.<p>` and inserting that?

Reply:
What about we keep it simple, and use `userAssert()` instead?

Reply:
Good idea. Done.